dxp/how_clay_arrives_to_dxp.md

```diff
@@ -0,0 +1,223 @@
+| Common Liferay components (Portlets)                      |          | ✅     | ✅      | ✅    |
+| Liferay customizations of Clay components                 |          | ✅     | ✅      | ✅    |
+| Compat Layer: compatibility layer for Bootstrap 2 and 3   |          | ✅     | ✅      | ✅    |
+| Main CSS of the Liferay websites and pages                |          |        | ✅      |       |
+| CSS Custom Properties Layer: Modern customization of Clay |          |        | ✅      |       |
+| Hello World Widget: Front Page Banner                     |          |        | ✅      |       |
+| Main CSS of the Liferay administration panels             |          |        |         | ✅    |
+
+From a more technical point of view, setting the base through the generator will create a setting named `liferayTheme` in the `package.json` file of the theme folder.
+
+```json
+{
+	"name": "my-liferay-theme",
+	"version": "1.0.0",
+	"main": "package.json",
+	"keywords": ["liferay-theme"],
+	"liferayTheme": {
+		"baseTheme": "styled",
+		"templateLanguage": "ftl",
+		"version": "7.0"
+	},
+	"devDependencies": {
+		"gulp": "^3.8.10",
+		"liferay-theme-deps-7.1": "*",
+		"liferay-theme-tasks": "*"
+	}
+}
+```
+
+[Source](https://github.com/liferay/liferay-js-themes-toolkit/tree/master/packages/liferay-theme-tasks#liferaytheme)
+
+## How to import Clay into a custom-theme?
+
+We always recommend to use [liferay-js-themes-toolkit](https://github.com/liferay/liferay-js-themes-toolkit) to create or update a theme, but we understand that sometimes you could have the need of adding Clay to an old heavy-customized theme that can't be automatically updated. In those cases we will have to import Clay as a depencency.
+
+The first step will be to localize the project's folder and install the right dependencies. You can either use the classic Node system or the modern Yarn one.
+
+Node
+`npm install @clayui/css`
+
+Yarn
+`yarn add @clayui/css`
+
+It's important to know that these commands will only install the CSS library, if you are building a theme that can make a good use of Clay components created in React, we recommend to use:
+
+Node
+`npm install @clayui/*`
+
+Yarn
+`yarn add @clayui/*`
+
+You can also address specific components with commands like `npm install @clayui/buttons`, `yarn add @clayui/cards`, `yarn add @clayui/tables@^version`, etc. [Read more at Clay's website](https://clayui.com/docs/get-started/how-to-use-clay.html)
+
+## How is Clay imported into a module?
+
+Unlike the previous section we won't need to install Clay as a dependency in a module inside Liferay Portal. Thanks to our rich environment, we already have it included and we just need to enable it.
+
+To do so, we need to define the new import into our dedicated buildCSS process, we can do that copy the following code into the `build.gradle` file of our module.
+
+```js
+buildCSS {
+	imports = [
+		new File(npmInstall.nodeModulesDir, "@clayui/css/src/scss")
+	]
+}
+```
```

Comment:
This is interesting, because we're bringing it transitively through modules, but provide it as a kind of `devDependency`...

@wincent wondering in the correctness of this and if we should add `@clayui/css` as a top-level dependency... 🤔

Reply:
> wondering in the correctness of this and if we should add `@clayui/css` as a top-level dependency... 🤔

The only gotcha I see there is that I believe the `@clayui/css` dependency can drift quite a bit across the branches (`7.1.x`, `7.2.x`, `7.3.x`, `master`).

Currently it's at v3.6.0, v3.6.0, v3.17.1, and v3.18.1, respectively, on those branches.

So, if we bring it in the "officially blessed way" of bringing in `devDependencies` (ie. via `@liferay/npm-scripts`) then that means that it may make it a little harder to backport updates to `npm-scripts`, because they will bring with them a relatively high-risk CSS change each time. Having said that, we have quite a bit of draft across branches with respect to `npm-scripts` versions too at the moment... (n/a, v27.0.0, v32.5.1, and v32.7.0, respectively). I'd like to get those a little bit more in sync before the next release, if possible.

Reply:
We do bring it in an officially blessed way because we only allow one version of `@clayui/css` which matches that of `frontend-taglib-clay`.

Reply:
![blessing](https://user-images.githubusercontent.com/7074/94679846-10777380-0321-11eb-8f0f-86a039d4116f.gif)

Reply:
Ah, FYI, I meant to add it as a `devDependency` in `modules/package.json`, not in `npm-scripts`.